native_client/javascript/client.js

```diff
@@ -44,6 +45,13 @@ function totalTime(hrtimeValue) {
   return (hrtimeValue[0] + hrtimeValue[1] / 1000000000).toPrecision(4);
 }
 
+const buffer = Fs.readFileSync(args['audio']);
+const result = Wav.decode(buffer);
+
+if (result.sampleRate < 16000) {
+  console.error('Warning: original sample rate (' + result.sampleRate + ') is lower than 16kHz. Up-sampling might produce erratic speech recognition.');
+}
+
 var audioStream = new MemoryStream();
 Fs.createReadStream(args['audio']).
```

Comment:
This is now reading the file twice. You can do this to reuse the existing buffer: https://gist.github.com/reuben/a8268789bdcc4dcfb5cecc61829848c9

Reply:
Awesome, I wanted to do that but I did fail.